README.md

```diff
@@ -709,6 +722,149 @@ options on to the custom `TaskQueue` implementation. (Note that because the
 queue options are set as a property on the task, tasks with queue
 options cannot be submitted as JavaScript primitives).
 
+## Custom Worker Scheduling
+
+Piscina provides a default scheduling algorithm that attempts to spawn threads
+based on the max number of threads, and the possibility of a task being cancelled.
+
+This algorithm can be overriden by passing a Custom Scheduler to the `Piscina`
+constructor.
+
+Custom schedulers must implement the `Scheduler` interface, described below:
+```ts
+class TaskScheduler {
+  size: number = 0
+
+  // The constructor will receive the max number of concurrent task per worker
+  // as passed when instantiating Piscina
+  constructor (concurrentTasksPerWorker: number) {
+  }
+
+  [Symbol.iterator] (): IterableIterator<PiscinaWorker> {
+    // Meant to enable Piscina iterate over the workers for different purposes
+  }
+
+  add (worker: PiscinaWorker): void {
+    // Will be called when a new worker is created
+    // Workers will be marked as ready after initialization
+    // You can listen to it calling worker.onReady and scheduling a callback
```

Comment:
When are workers created? Shouldn't that be up to the scheduler?

Reply:
It's kind of hard to leverage this to the Scheduler, as Piscina does pretty much several abstractions for the threads when spawned.
Overall, the Scheduler can indicate to Piscina to spawn a new thread by returning null when calling `Scheduler#pick`.